### DISCUSSION TITLE
"Docker Compose up airflow-init" is running extremely slowly, with fast downloads for Postgres and Redis, but slow for airflow-init. How can I resolve this?

### DISCUSSION THREAD

**Discussion by shark8848** [apache/airflow#34152] [Q&A]
"Docker Compose up airflow-init" is running extremely slowly, with fast downloads for Postgres and Redis, but slow for airflow-init. How can I resolve this?

**Comment by Taragolis** [comment]
Could you clarify, which docker compose you run? If possible attach it. Your operation system also could be helpful for someone who would try to help you with problem. More details better then less details

I've just check on latest [**sample** docker compose](https://airflow.apache.org/docs/apache-airflow/2.7.0/docker-compose.yaml), and it run pretty quick in case if docker images already pulled. On my machine (Linux) it takes for about 20 seconds, to pass all healthchecks and start last containers

**Comment by potiuk** [comment]
What exactly is running slow for you ? can you elaborate what part of the docker compose is slow for you ? Screenhots, videos and examples would be good. IT might be your network for example - Airlfow's reference images are quite big as they contain many providers that users might want to use and their system dependncies, but you can build and use your own image with only what you need  https://airflow.apache.org/docs/docker-stack/build.html and optimize it as you want. 

You can also just store the image in your local registry if your connection to dockerhub registry is slow.

But this is just guess that it's the image pull that is slow - without your explanation of what is "slow" it's impossible to say.


You might also have use old version of Airflow  image and run it on MacOS with ARM /M1/M2  (airflow supports ARM images natively as of Airflow 2.3 - see the changelog: https://airflow.apache.org/docs/docker-stack/changelog.html#airflow-2-3 before that Docker would automatically use emulation - which is indeed around 16 x slower. But you will see it in your Docker Desktop as container running with emulation.

BTW. @Taragolis - this time you beat me to it :)

**Reply by Taragolis** [reply]
Your suggestion has more detail
